src/pudl_archiver/archivers/ferc/xbrl.py

```diff
@@ -108,11 +115,47 @@ def parse_timestamp(cls, timestamp: time.struct_time):  # noqa: N805
         return datetime.datetime.fromtimestamp(time.mktime(tuple(timestamp)))
 
     def __hash__(self):
-        """Implement hash so FeedEntry can be used in a set.
+        """Implement hash so FeedEntry can be used in a set."""
+        return hash(f"{self.download_url}")
 
-        Entry ID's are unique, so that's all that is needed for the hash.
-        """
-        return hash(f"{self.entry_id}")
+    def __eq__(self, other: "FeedEntry"):
+        """Implement eq so FeedEntry can be used in a set."""
+        return self.download_url == other.download_url
+
+
+def _taxonomy_zip_name_from_url(url: str) -> str:
+    if not (match := TAXONOMY_URL_PATTERN.match(url)):
+        raise RuntimeError(f"{url} does not appear to be a taxonomy url.")
+    return f"{match.group(1).replace('_', '-')}.zip"
+
+
+class FilingMetadata(BaseModel):
+    """Combines RSS feed metadata with taxonomy referenced in filing."""
+
+    filename: str
+    rss_metadata: FeedEntry
+    taxonomy_url: str
+    taxonomy_zip_name: str
+
+    @classmethod
+    def from_rss_metadata(
+        cls, rss_metadata: FeedEntry, filename: str, filing_data: bytes
```

Comment:
Seems like `filename` is unused in the function body, and we can remove this parameter.

Reply:
Aha! Now it is being used.